air/src/boundry.rs

```diff
@@ -0,0 +1,271 @@
+use ark_ff::PrimeField;
+use composition_polynomial::{air::Air, composition_polynomial::CompositionPolynomial};
+
+#[derive(Clone)]
+pub struct BoundaryAir<F: PrimeField> {
+    trace_length: usize,
+    n_columns: usize,
+    constraints: Vec<ConstraintData<F>>,
+    mask: Vec<(usize, usize)>,
+}
+
+#[derive(Clone)]
+pub struct ConstraintData<F> {
+    coeff_idx: usize,
+    column_index: usize,
+    point_x: F,
+    point_y: F,
+}
+
+impl<F: PrimeField> BoundaryAir<F> {
+    pub fn new(
+        trace_length: usize,
+        n_columns: usize,
+        boundary_conditions: &[(usize, F, F)],
+    ) -> Self {
+        let mut constraints = Vec::with_capacity(boundary_conditions.len());
+
+        for (coeff_idx, &(column_index, ref point_x, ref point_y)) in
+            boundary_conditions.iter().enumerate()
+        {
+            let x = *point_x;
+
+            let pos = constraints
+                .iter()
+                .position(|constraint: &ConstraintData<F>| constraint.point_x == x);
```

Comment:
Might be fine since the number of boundary constraints is probably small, but this has O(n^2) complexity.

Maybe use a Map : point_x -> ConstraintData

Reply:
What is the use case of "overwriting" a previous boundary constraint with a different point_y? Should we just error instead?